python/ray/experimental/serve/task_runner.py

```diff
@@ -0,0 +1,86 @@
+import traceback
+
+import ray
+
+
+class TaskRunner:
+    """A simple class that runs a function.
+
+    The purpose of this class is to model what the most basic actor could be.
+    That is, a ray serve actor should implement the TaskRunner interface.
+    """
+
+    def __init__(self, func_to_run):
+        self.func = func_to_run
+
+    def __call__(self, *args):
+        return self.func(*args)
+
+
+def wrap_to_ray_error(callable_obj, *args):
+    """Utility method that catch and seal exceptions in execution"""
+    try:
+        return callable_obj(*args)
+    except Exception:
+        traceback_str = traceback.format_exc()
+        return ray.exceptions.RayTaskError(str(callable_obj), traceback_str)
+
+
+class RayServeMixin:
+    """This mixin class adds the functionality to fetch from router queues.
+
+    Warning:
+        It assumes the main execution method is __call__ of the current class.
+
+    Example:
+        >>> # to make MyClass servable
+        >>> @ray.remote
+            class RayServeActor(MyClass, RayServeMixin):
```

Comment:
Isn't this in the wrong order? Python inheritance is right to left, so in this case the user could overwrite the serve methods. Instead, we should throw an error if we're overwriting theirs.

Reply:
I'm not a fan of this API in general. Why not add a new decorator to keep it consistent (e.g., `@ray.serve`)? This would also give us more flexibility in how we handle the definition.

Reply:
I namespaced all instance variables and method name with `_ray_serve_`. A new decorator could work but it's more heavy weight. I would prefer to leave it for future work. In addition, this inheritance hierarchy is not directly exposed to the user.

Reply:
I suppose it's fine for an internal API. The inheritance order is still wrong, though.